### ISSUE TITLE
Need a JSON schema for our configuration parameters

### ISSUE THREAD

**Issue by moxiegirl** [blockstack/gaia#202]
I manually created one but I doubt it is complete.  @kantai 

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "servername": { "type": "string" },
    "port": { "type": "integer" },
    "driver": { "type": "string" },
    "bucket": { "type": "string" },
    "readURL": { "type": "string" },
    "requireCorrectHubUrl": { "type": "string" },
    "awsCredentials": {
      "type": "object",
      "properties": {
        "endpoint": { "type": "string"},
        "accessKeyId": { "type": "string"},
        "secretAccessKey": { "type": "string"}
      },
      "required": [
        "endpoint",
        "accessKeyId",
        "secretAccessKey"
      ]
    },
    "proofsConfig": {
      "type": "object",
      "properties": {
        "proofsRequired": { "type": "string"}
      },
      "required": [
        "proofsRequired"
      ]
    },
    "azCredentials": {
      "type": "object",
      "properties": {
        "accountName": { "type": "string"},
        "accountKey": { "type": "string"}
      },
      "required": [
        "accountName",
        "accountKey"
      ]
    },
    "gcCredentials": {
      "type": "object",
      "properties": {
        "keyFilename": { "type": "string"}
      },
      "required": [
        "keyFilename"
      ]
    },
    "diskSettings": {
      "type": "object",
      "properties": {
        "storageRootDirectory": { "type": "string"}
      },
      "required": [
        "storageRootDirectory"
      ]
    },
    "pageSize": { "type": "integer"},
    "argsTransport": {
      "type": "object",
      "properties": {
        "level": { "type": "string"},
        "handleExceptions": { "type": "boolean" },
        "stringify": { "type": "boolean" },
        "timestamp": { "type": "boolean" },
        "colorize": { "type": "boolean" },
        "json":{ "type": "boolean" }
      },
      "required": [
        "level",
        "handleExceptions",
        "stringify",
        "timestamp",
        "colorize",
        "json"
      ]
    }
  },
  "required": [
    "servername",
    "port",
    "driver",
    "bucket",
    "readURL",
    "requireCorrectHubUrl",
    "awsCredentials",
    "proofsConfig",
    "azCredentials",
    "gcCredentials",
    "diskSettings",
    "pageSize",
    "argsTransport"
  ]
}
```

**Comment by moxiegirl** [comment]
Waaaah. I assigned something.

**Comment by markmhx** [comment]
@kantai Are you actually currently working on this or should we move it out of "in progress" to the backlog or icebox?

**Comment by zone117x** [comment]
This issue should be closed via https://github.com/blockstack/gaia/pull/224

**Comment by kantai** [comment]
> This issue should be closed via #224

Yep
